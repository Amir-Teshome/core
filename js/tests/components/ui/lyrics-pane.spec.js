import Lyrics from '@/components/ui/lyrics-pane.vue'
import factory from '@/tests/factory'

describe('components/ui/lyrics', () => {
  it('displays lyrics if the song has lyrics', () => {
    const song = factory('song')
    expect(shallow(Lyrics, {
      propsData: { song }
    }).html()).toMatch(song.lyrics)
  })

  it('displays a fallback message if the song has no lyrics', () => {
    expect(shallow(Lyrics, {
      propsData: {
        song: factory('song', { lyrics: '' })
      }
    }).html()).toMatch('No lyrics found. Are you not listening to Bach?')
  })
})
