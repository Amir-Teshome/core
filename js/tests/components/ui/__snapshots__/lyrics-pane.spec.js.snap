// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ui/lyrics displays a fallback message if the song has no lyrics 1`] = `
<article id="lyrics">
  <div class="content">
    <div style="display: none;">
      <div></div>
      <div class="text-zoomer"><button title="Zoom out"><i class="fa fa-search-minus"></i></button> <button title="Zoom in"><i class="fa fa-search-plus"></i></button></div>
    </div>
    <p class="none">No lyrics found. Are you not listening to Bach?</p>
  </div>
</article>
`;

exports[`components/ui/lyrics displays lyrics if the song has lyrics 1`] = `
<article id="lyrics">
  <div class="content">
    <div>
      <div>Foo and bar</div>
      <!---->
    </div>
    <!---->
  </div>
</article>
`;
